src/adstaylorinteg.jl

```diff
@@ -0,0 +1,326 @@
+exp_model(t, p) = p[1] * exp(p[2] * t)
+exp_model!(F, t, p) = (@. F = p[1] * exp(p[2] * t))
+
+# In place jacobian of exp_model! wrt parameters p
+function exp_model_jacobian!(J::Array{T, 2}, t, p) where {T <: Real}
+    @. J[:, 1] = exp(p[2] * t)
+    @. @views J[:, 2] = t * p[1] * J[:, 1]
+end
+
+"""
+    size_per_variable(P::TaylorN{T}) where {T <: Real}
+    size_per_variable(P::AbstractVector{TaylorN{T}}) where {T <: Real}
+
+Return the truncation error of each jet transport variable in `P`.
+
+!!! reference
+    See section 3 of https://doi.org/10.1007/s10569-015-9618-3.
+"""
+function size_per_variable(P::TaylorN{T}) where {T <: Real}
+    # Jet transport order
+    varorder = P.order
+    # Number of variables
+    nv = get_numvars()
+    # Absolute sum per variable per order
+    ys = zeros(T, nv, varorder+1)
+
+    for i in eachindex(P.coeffs)
+        idxs = TaylorSeries.coeff_table[i]
+        for j in eachindex(idxs)
+            for k in eachindex(idxs[j])
+                ys[k, idxs[j][k]+1] += abs(P.coeffs[i].coeffs[j])
+            end
+        end
+    end
+    # Initial parameters
+    p0 = ones(T, 2)
+    # Orders
+    xs = 0:varorder
+
+    M = Vector{T}(undef, nv)
+    for i in eachindex(M)
+        # Non zero coefficients
+        idxs = findall(!iszero, view(ys, i, :))
+        # Fit exponential model
+        fit = curve_fit(exp_model!, exp_model_jacobian!, view(xs, idxs),
+                        view(ys, i, idxs), p0; inplace = true)
+        # Estimate next order coefficient
+        exp_model!(view(M, i:i), varorder+1, fit.param)
+    end
+
+    return M
+end
+
+function size_per_variable(P::AbstractVector{TaylorN{T}}) where {T <: Real}
+    # Number of variables
+    nv = get_numvars()
+    # Size per variable per element of P
+    norms = Matrix{T}(undef, nv, length(P))
+    for i in eachindex(P)
+        norms[:, i] .= size_per_variable(P[i])
+    end
+    # Sum over element of P
+    return sum(norms, dims = 2)
+end
+
+"""
+    splitdirection(P::TaylorN{T}) where {T <: Real}
+    splitdirection(P::AbstractVector{TaylorN{T}}) where {T <: Real}
+
+Return the index of the jet transport variable with the largest truncation error
+according to [`size_per_variable`](@ref).
+"""
+function splitdirection(P::TaylorN{T}) where {T <: Real}
+    # Size per variable
+    M = size_per_variable(P)
+    # Variable with maximum error
+    return argmax(M)[1]
+end
+
+function splitdirection(P::AbstractVector{TaylorN{T}}) where {T <: Real}
+    # Size per variable
+    M = size_per_variable(P)
+    # Variable with maximum error
+    return argmax(M)[1]
+end
+
+"""
+    adsnorm(P::TaylorN{T}) where {T <: Real}
+
+Return the truncation error of `P`.
+"""
+function adsnorm(P::TaylorN{T}) where {T <: Real}
+    # Jet transport order
+    varorder = P.order
+    # Absolute sum per order
+    ys = Vector{T}(undef, varorder+1)
+
+    for i in eachindex(ys)
+        ys[i] = sum(abs, P.coeffs[i].coeffs)
+    end
```

Comment:
Suggestion: does it make sense here to pass `ys` as an arg to `adsnorm` so that `ys` is only updated here without having to allocate it on every call?